plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -899,6 +899,9 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
         ImmutableMap.Builder<String, Optional<Partition>> resultBuilder = ImmutableMap.builder();
         for (Entry<String, List<String>> entry : partitionNameToPartitionValuesMap.entrySet()) {
             Partition partition = partitionValuesToPartitionMap.get(entry.getValue());
+            if (partition == null) {
```

Comment:
This logging strategy may be rather naive. 
If lots of partitions are deleted between the time their names gets retrieved and the time their metadata gets retrieved, we could end up with lots of useless warnings in the logs.